azkaban-exec-server/src/main/java/azkaban/execapp/FlowPreparer.java

```diff
@@ -107,6 +121,9 @@ void setupProject(final ProjectVersion pv)
     // If directory exists. Assume its prepared and skip.
     if (pv.getInstalledDir().exists()) {
       log.info("Project already cached. Skipping download. " + pv);
+      touchIfExists(
+          new File(Paths.get(pv.getInstalledDir().getPath(), Constants.PROJECT_DIR_SIZE_FILE_NAME)
```

Comment:
Why not use Path as the argument to touchIfExists? File is a legacy class. Path is preferred now if I remember correctly.

Reply:
https://gquintana.github.io/2017/09/02/Java-File-vs-Path.html

Reply:
good point, updated. thanks